# Probability of die showing a number.

Event E: Probability of a die showing either 1 or 4 when thrown

Import necessary packages

In [ ]:
import numpy as np

# Function for uniform outcome

$n$: number of outcomes in the sample space

Output: $m$ outcomes selected uniformly at random from 1 to $n$


In [ ]:
def uniform(n,m):
  return np.random.randint(1,n+1,size=m)

# Probability that the number shown after throwing a die is 1 or 4.

## Manual Calculations.

We have a sample space S={1,2,3,4,5,6} where all the outcomes are equally likely. Probability that the outcome is 1, P(A)=1/6 and for 4, P(B)=1/6. Since these are independent events probability that event E, that the outcome is either 1 or 4, occurs is P(E)=P(A)+P(B)

i.e. P(E)=2/6=0.3333

## Monte Carlo simulation

We will use the Monte Carlo simulation to find the probability that a die shows either 1 or 4.The estimate is close to $P(E)$. Results are more consistent with high number of repetitions. Here we are simulating it 10000 times.

In [7]:
no = 0   #variable for storing the number of times the event occurs
for i in range(10000): #Total number of repetitions
  die = uniform(6,1)  #experiment
  if die == 4 or die == 1: #Event
    no = no + 1
print(no/10000) #probability estimate by Monte Carlo

0.3365


# Analysis

Clearly, the probability calculated by the Monte Carlo simulation, 0.3365, for the event is very close to the probability that was calculated manually, which is 0.333.